usecase/csv/csv.go

```diff
@@ -0,0 +1,49 @@
+package csv
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+	"bootcamp/domain/model"
+	"bootcamp/utils"
+	"github.com/gorilla/mux"
+)
+
+/*
+GetPokemon returns a Pokemon list read from a CSV file
+If query params exists or /{id}, filter the Pokemon
+Otherwise, returns the list
+*/
+func GetPokemon(r *http.Request) (model.PokemonList, error) {
+	pokemonList, err := utils.ReadCSV()
+	var pokemonSubset model.PokemonList
+	params := mux.Vars(r)
```

Comment:
The usecase layer shouldn't be dealing with `Request` information, the controller should read all the information it needs from the request, validate it, and then sending the validated information to the usecase

Reply:
*